lazyflow/operators/ioOperators/opInputDataReader.py

```diff
@@ -152,21 +165,18 @@ def setupOutputs(self):
         Inspect the file name and instantiate and connect an internal operator of the appropriate type.
         TODO: Handle datasets of non-standard (non-5d) dimensions.
         """
-        filePath = self.FilePath.value
-        assert isinstance(filePath, (str, unicode)), "Error: filePath is not of type str.  It's of type {}".format(
-            type(filePath)
-        )
-
-        # Does this look like a relative path?
-        useRelativePath = not isUrl(filePath) and not os.path.isabs(filePath)
-
-        if useRelativePath:
-            # If using a relative path, we need both inputs before proceeding
-            if not self.WorkingDirectory.ready():
-                return
+        path_components = splitPath(self.FilePath.value)
+
+        cwd = self.WorkingDirectory.value if self.WorkingDirectory.ready() else None
```

Comment:
```python
if not self.WorkingDirectory.ready():
    return  # FIXME: this mirrors old logic but I'm not sure if it's safe

cwd = self.WorkingDirectory.value
path_components = splitPath(self.FilePath.value)
filePath = os.path.pathsep.join(make_absolute(p, cwd) for p in path_components)
```

Reply:
I really dislike this `WorkingDirectory` slot here -.- , but this is not really the same logic as the one in the PR; If this method only got absolute paths in its `FilePath` input, then it'd be fine even without a `ready()` `WorkingDirectory` slot.